Firestore/Example/FuzzTests/FSTFuzzTestsPrincipal.mm

```diff
@@ -49,27 +59,83 @@ int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
   return 0;
 }
 
+// Retrieves the fuzzing target from the FUZZING_TARGET environment variable.
+// Default target is NONE if the environment variable is empty, not set, or
+// could not be interpreted. Should be kept in sync with FuzzingTarget.
+FuzzingTarget GetFuzzingTarget() {
+  NSString *fuzzing_target_env =
+      [[[NSProcessInfo processInfo] environment] objectForKey:@"FUZZING_TARGET"];
+
+  if (fuzzing_target_env == nil || [fuzzing_target_env length] == 0) {
+    LOG_WARN("No value provided for FUZZING_TARGET environment variable.");
+    return NONE;
+  }
+
+  if ([@"NONE" isEqualToString:fuzzing_target_env]) {
+    return NONE;
+  }
+
+  if ([@"SERIALIZER" isEqualToString:fuzzing_target_env]) {
+    return SERIALIZER;
+  }
+
+  // Value did not match any target.
+  LOG_WARN("Invalid fuzzing target: %s", fuzzing_target_env);
+
+  // Default is NONE.
+  return NONE;
+}
+
 // Simulates calling the main() function of libFuzzer (FuzzerMain.cpp).
+// Uses GetFuzzingTarget() to get the fuzzing target and sets libFuzzer's args
+// accordingly.
 int RunFuzzTestingMain() {
-  // Get dictionary file path from resources and convert to a program argument.
+  // Get the fuzzing target.
```

Comment:
Optional nit: I think lines 93-103 are closely related and would look better if squashed closer together.

Reply:
Done.